### DISCUSSION TITLE
Velocità e latenze di risposta di Home Assistant

### DISCUSSION THREAD

**Discussion by giancaos** [indomus/forum#981] [GENERALE - Discussioni generiche]
Ciao a tutti,

Ho un sistema domotico misto che gira da un paio di anni con un Hub Domoticz.
Misto nel senso che ho alcuni Shelly, degli Zigbee (zigbee2mqtt) e un sistema custom su 485 vecchio di 20 anni che carica gli eventi tramite MQTT e che veicola il 100% di pulsanti e luci di tutta la casa.

Le logiche (del tipo: leggo quell'input e accendo quell'output) sono assolti da scripts in DZ-VENTS.
Non tutte per fortuna, in quanto il sistema su 485 ha delle logiche sue e assolve molti pulsanti e molte lampade in casa.
Domoticz gira su una Beaglebone Black.
Ho notato che le logiche assolte da Domoticz (e quindi dagli scripts) non sono un granché in termini di velocità di esecuzione, soprattutto se una delle periferiche coinvolte e' Zigbee.

Dopo due anni ho pensato di fare un upgrade al sistema, questo consiste in:
- sostituire la BeagleBone con una Banana PI M2 (ARM a 4 core e file sistem su eMMC). 
- sono combattuto, e qui il fulcro delle ai domanda, nel sostituire Domoticz con Home Assistant
Va detto che a suo tempo scelsi Domoticz in quanto sviluppato in C/C++ contro gli altri HUB che erano in Java o linguaggio piu' ad alto livello.

Adlila' che devo ancora capire come in Home Assistant vanno risolte le logiche di attivazione ma secondo voi potrei incorrere ad un tira e molla, ossia: migliorare l'hardware ma poi perdere in velocità di esecuzione delle attivazioni per il cambio di hub?

Grazie per eventuali osservazioni e risposte.
bye
g

**Comment by snakuzzo** [comment]
Home Assistant Core (non HassOS) non ha bisogno di nulla di particolare per girare. È scritto in Python ed è efficientissimo.

Mai sentito nessuno lamentarsi di lentezza nell'attivazione di qualcosa su HA. 

Puoi tranquillamente installarlo (magari con docker) e tenerlo up&running assieme a Domoticz per provare

**Reply by giancaos** [reply]
Grazie dell'intervento. Per lentezza intendo anche un secondo. Il mio sistema demotico e' integrato, riguarda tutti i pulsanti delle luci di casa, ad esempio. Entrare in un stanza e premere un pulsante per accendere la luce un secondo di tempo potrebbe significare anche il disagio di fare qualche passo e poi tendere a tornare indietro per ri-pigiare in quanto non si ha avuto l'effetto immediato. Un ritardo di un secondo potrebbe essere significativo se ci pensi.

grazie ancora
bye
g

PS
Ho dato una sistemata al thread, francamente non sapevo che fosse una struttura a subthread, per quello non ti ho risposto a dovere.

**Reply by snakuzzo** [reply]
Ti ripeto...mai avuto lentezza lato HA...
Poi magari se dovessi averla andrebbe vista la causa...ma non credo proprio che possa dipendere da HA.
C'è gente che usa HA anche per la gestione di alberghi con centinaia di entità e problemi di questo tipo non ne ha mai avuti.

**Reply by giancaos** [reply]
Ho appena finito ad installarlo in modalità core. La stessa macchina dove gira e' dotata di Mosquitto. Vorrei tenere totalmente separati i due servizi (Domoticz/HA). Ora devo implementare un doppio servizio client di MQTT nel mio vecchio sistema custom in modo che assolva entrambi gli hub. Cosi' se HA mi convincerà poi dovrò solo togliere la BeagleBone. 
AL momento noto che HA integra un sacco di cose (mi ha visto il televisore, i Sonos e perfino il server piHole), e' impressionante, siamo avanti anni luce rispetto Domoticz.

Grazie ancora. Eventuali feedback saranno riportati nel thread che ho aperto.
bye
g

**Reply by snakuzzo** [reply]
Il broker resta lo stesso e lato client hai già tutto su HA. Devi solo usare in client id diverso per HA

**Reply by giancaos** [reply]
Si si, certo, so che lo posso fare. Pero' ho creato un altro broker sulla macchina di HA, volevo rendermi totalmente indipendente dalla BeagleBone dove gira l'altro Broker per poter un giorno rimuovere in maniera del tutto indolore Domoticz e relativa macchina. Non ho grossi problemi ad implementare una doppia pubblicazione sul sistema custom, e' roba in C che ho fatto io e la conosco bene.

Paradossalmente devo capire come si fa una cosa apparentemente piu' facile, ossia fare il subscribe su HA del topic  MQTT del mio sistema e analizzare il payload e associarlo ad un evento.
Su Domoticz era relativamente semplice, bastava pubblicare in tracciatura jason l'id dell'oggetto e il suo nuovo valore (booleano/int/float). Poi uno script captava l'evento e li' sviluppavi il da farsi.
Ad ogni modo indomus.it e' straordinario in questo senso, trovi tutti i tutorial possibili e anche di piu'.

bye
g

**Reply by snakuzzo** [reply]
Fai come credi, ma continuo a ripetere che non serve a niente e ti complichi solo le cose.
Sul come creare le entità basate su MQTT è molto semplice. Leggi la doc e vedi che troverai tutto

**Reply by giancaos** [reply]
> Fai come credi, ma continuo a ripetere che non serve a niente e ti complichi solo le cose.

Le circostanze in campo le ho presente solo io, ad ogni modo ho gia' finito le patches, appena possibile rifaccio il boot con il nuovo software (devi pensare che stiamo parlando di un sistema a bus dove ogni pulsante, ogni punto luce e tapparelle, riscaldamento e antiintrusione sono operative in un'intera casa ABITATA). Capisco comunque il tuo consiglio, e' che a volte la strada piu' semplice non e' sempre la migliore.
Nel frattempo riconosco in HA un progetto spaziale, continuo a scoprirne aspetti veramente interessanti che sfrutterò a dovere. Su Banana PI dotato di Arm a 4 core sembra volare, vedremo a pieni giri come andrà... 

bye
g

**Reply by snakuzzo** [reply]
Quello che volevo dire è...
Se tutto quello che hai dentro cada utiilizzq il protocollo MQTT, hai tutto già pronto. Ti basta collegare HA allo stesso broker e farlo presentare con un client id diverso da quello di domoticz (e questo ti viene già gratis). Poi ti basta solo creare le entità con platform mqtt su HA utilizzando topic e payload allo stesso modo. Così facendo piloti gli stessoi "device" senza creare conflitti tra i due hub ed hai lo stato aggiornato in tempo reale dappertutto. Non c'è alcun rischio di creare casini...non so se ti è chiaro...e questo vale a prescindere dal tipo di situazione che hai a casa

**Comment by giancaos** [comment]


**Reply by snakuzzo** [reply]
Innanzi tutto ti chiedo la cortesia di rispondere "bene" senza creare nuovi thread, ma rispondendo sotto l'ultimo messaggio (a meno che non passi a parlare d'altro).
Per quanto riguarda il ritardo di un secondo, non credo proprio che HA abbia un ritardo del genere. Se proprio ci sarà un ritardo magari dipenderà dal device

P.S. per cortesia modifica il titolo di questa discussione ed utilizza un titolo un attimino più "serio". Grazie
